src/app/learnocaml_partition_view.ml

```diff
@@ -34,10 +33,27 @@ let open_tok tok =
   let _win = window_open ("/student-view.html?token="^tok) "_blank" in
   false
 
+(* let list_of_nick = 
+  List.map @@ fun x ->
+    let tok = Token.to_string x in
+    let nick = "toto" in (*trouver comment récupérer les pseudos*)
+    H.a ~a:[H.a_onclick (fun _ -> open_tok tok)] [H.txt (nick ^ " ")]
+
 let list_of_tok =
   List.map @@ fun x ->
     let tok = Token.to_string x in
-    H.a ~a:[H.a_onclick (fun _ -> open_tok tok)] [H.txt (tok ^ " ")]
+    H.a ~a:[H.a_onclick (fun _ -> open_tok tok)] [H.txt (tok ^ " ")] *)
+
+let selected_list students= 
+  List.map @@ fun x ->
+    let tok = Token.to_string x in
+    let choice = Manip.value (find_component "Hide tokens") in
+    match choice with
+      |"tokens" -> 
+        H.a ~a:[H.a_ondblclick (fun _ -> open_tok tok)] [H.txt (tok ^ " ")]
+      |"nicknames" ->
+        let nick = students.nickname in (*trouver comment récupérer les pseudos*)
+        H.a ~a:[H.a_ondblclick (fun _ -> open_tok tok)] [H.txt (nick ^ " ")]
```

Comment:
I have another algorithm to suggest, that may easily avoid "needing to reload the page every time one updates the dropdown":

* You could add (at load time from js_of_ocaml) three consecutive fields for each student identifier, e.g.
  `<div class="anon-id">1</div><div class="token-id">123-456-789-ABC</div><div class="nickname-id">John</div>`
* And ensure that for the event `Manip.Ev.onchange_select (find_component "learnocaml-select-student-info")`,
  some dynamic code just adds one class among {anon-id, token-id, nickname-id} to the two surrounding div's
  `<div id="learnocaml-exo-tab-list">` and `<div id="learnocaml-exo-tab-details">`
* And finally configuring the CSS so that any inner div such as
  `<div id="learnocaml-exo-tab-list" class="nickname-id">…<div class="token-id">123-456-789-ABC</div>…</div>`
  is hidden (`display: none`) because "token-id" ≠ "nickname-id".

What about trying this?